### DISCUSSION TITLE
Error : Joomla vote plugin ItemReviewed field is missing

### DISCUSSION THREAD

**Discussion by ManuelVoileux** [joomla/joomla-cms#42920] [Q&A]
Hi, 
I have activated the the core Joomla vote on content plugin. It works OK on front end. But When I check with the Google validating tool https://search.google.com/test/rich-results?hl=en, I get a critical error. itemReviewed field is missing. 

I guess that the file /plugins/content/vote/tmpl/rating.php  should be adjusted. 

PMLeconte on the French community Joomla forum has posted a partial answer. https://forum.joomla.fr/forum/joomla-5-x/questions-g%C3%A9n%C3%A9rales-ab/2060593-erreur-indexation-google-vote-sur-article?p=2060635#post2060635

conseilgouz proposes an extra-vote plugin from JoomlaHill  https://github.com/conseilgouz/pkg_extravote_j4

How can this be generalized ?  I am not a coder, and I cannot cope with php.

How can this be solved ? Thanks for help

![Screenshot 2024-02-28 at 10-40-17](https://github.com/joomla/joomla-cms/assets/117023571/95784339-3912-4be4-9b8e-df65363adc8b)

**Comment by ManuelVoileux** [comment]
With the following code from [ConseilGouz](https://github.com/conseilgouz) , in the rating.php file, we get the itemReviewed field. The itemscope is set to Recipe , as it was tested on a Recipe article.
```
<div class="visually-hidden" itemprop="aggregateRating" itemscope itemtype="https://schema.org/AggregateRating">
            <div itemprop="itemReviewed" itemscope itemtype="https://schema.org/Recipe">
                      <span itemprop="name"><?php echo $row->title;?></span>
            </div>
            <?php echo Text::sprintf('PLG_VOTE_USER_RATING', '<span itemprop="ratingValue">' . $rating . '</span>', '<span itemprop="bestRating">5</span>'); ?>
            <meta itemprop="ratingCount" content="<?php echo $rcount; ?>">
            <meta itemprop="worstRating" content="1">
</div>
```

But Google sees a recipe and shows a critical error as it wants all the items for a recipe, the minimum is an image. With the following code it gets an image : 
```
$monimage = Uri::root()."images/powered_by.png"; // mettre votre image par défaut
$recipe_images = json_decode($row->images);
if (!empty($recipe_images->image_intro)) { // remove # info after image
    $monimage = explode('#',$recipe_images->image_intro);
    $monimage = Uri::root().$monimage[0];} 
<div class="visually-hidden" itemprop="aggregateRating" itemscope itemtype="https://schema.org/AggregateRating">
<div itemprop="itemReviewed" itemscope itemtype="https://schema.org/Recipe">
   <span itemprop="name"><?php echo $row->title;?></span>
   <span itemprop="image"><?php echo $monimage;?></span>
</div>
<?php echo Text::sprintf('PLG_VOTE_USER_RATING', '<span itemprop="ratingValue">' . $rating . '</span>','<span itemprop="bestRating">5</span>'); ?>
   <meta itemprop="ratingCount" content="<?php echo $rcount; ?>">
  <meta itemprop="worstRating" content="1">
</div>
```

But Then Google sees 2 Recipes : the one indexed with the Schema tab, and the one indexed with the rating.php  On top of that , the rating .php is tuned for Recipe, not for every type. It is necessary to merge both to get only one indexed article as Recipe, and to get the type from the schemaorg tab.

**Comment by conseilgouz** [comment]
As suggested on the french forum, try http://schema.org/Product instead of recipe.

According to Schema.org doc https://schema.org/Product , it behaves like a generic type.

Suggested plugin override,  replacing line 82 to  86,  becomes :

        <div class="visually-hidden" itemscope="itemscope" itemtype="https://schema.org/Product">
            <span itemprop="name"><?php echo $row->title;?></span>
            <div class="visually-hidden" itemprop="aggregateRating" itemscope itemtype="https://schema.org/AggregateRating">
            <?php echo Text::sprintf('PLG_VOTE_USER_RATING', '<span itemprop="ratingValue">' . $rating . '</span>', '<span itemprop="bestRating">5</span>'); ?>
            <meta itemprop="ratingCount" content="<?php echo $rcount; ?>">
            <meta itemprop="worstRating" content="1">
        </div>
        </div>​

**Comment by ManuelVoileux** [comment]
Thanks for your work. It works fine
![Screenshot 2024-03-01 at 13-30-10](https://github.com/joomla/joomla-cms/assets/117023571/fa3362d7-398c-4394-9733-e4fca97437df)

**Comment by conseilgouz** [comment]
To make your test a "human test", please use https://issues.joomla.org/tracker/joomla-cms/42933

To make this PR RTC, after code reviews and comments from Dev Team, we need at least 2 "human tests".

Pascal
